arangod/Agency/Supervision.cpp

```diff
@@ -1276,7 +1279,107 @@ void Supervision::workJobs() {
       .run(dummy);
     LOG_TOPIC("99006", TRACE, Logger::SUPERVISION) << "Finish JobContext::run()";
   }
+}
+
+
+bool Supervision::verifyCoordinatorRebootID(std::string coordinatorID, uint64_t wantedRebootID) {
+  // check if the coordinator exists in health
+  std::string const& health = serverHealth(coordinatorID);
+  // if the server is not found, health is an empty string
+  if (health != "GOOD" && health != "BAD") {
+    return false;
+  }
+
+  // now lookup reboot id
+  std::pair<uint64_t, bool> rebootID = _snapshot.hasAsUInt(curServersKnown + coordinatorID + "/" + StaticStrings::DatabaseCoordinatorRebootId);
+  return rebootID.second && rebootID.first == wantedRebootID;
+}
+
+void Supervision::deleteBrokenDatabase(std::string const& database, std::string const& coordinatorID, uint64_t rebootID) {
+  auto envelope = std::make_shared<Builder>();
+  {
+    VPackArrayBuilder trxs(envelope.get());
```

Comment:
I cannot get over this,
why are we creating it this complicated everytime and there is no:
builder = startEnvelope();
incPlanVersion(builder);
deleteDatabase(builder);
closeEnvelope(builder);
which would be 1.  much easier to understand, and 2. easier to maintain, 3. harder to get wrong...